test_launch_ros/test/test_launch_ros/descriptions/test_parameter_file.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for launch_ros.descriptions.ParameterFile."""
+
+from contextlib import contextmanager
+import os
+from tempfile import NamedTemporaryFile
+
+from launch import Substitution
+from launch.frontend import expose_substitution
+from launch.utilities import perform_substitutions
+
+from launch_ros.descriptions import ParameterFile
+
+import pytest
+
+
+@contextmanager
+def get_parameter_file(contents, mode='w'):
+    f = NamedTemporaryFile(delete=False, mode=mode)
+    try:
+        with f:
+            f.write(contents)
+        yield f.name
+    finally:
+        os.unlink(f.name)
```

Comment:
@ivanpauno nit: out of curiosity, why not:

```python
with NamedTemporaryFile() as f:
    f.write(contents)
    yield f.name
```
instead?

Reply:
that won't unlink the file, just close it (unlink = remove from the filesystem).

Reply:
It will remove it if you don't set `delete=False` (which seems unnecessary in this case).

Reply:
ah yes, I guess that works

Reply:
Done in https://github.com/ros2/launch_ros/pull/168/commits/fc999d0cc84d59e90a273e45cd1b0f268d2c0c58.

Reply:
I now remember why I did this ...

Windows doesn't allow opening the same file twice.
If I `yield` from within the `with` statement, then I have a problem when opening the file in the test (see Windows CI results).

Reply:
> Windows doesn't allow opening the same file twice.

That makes no sense.

Reply:
Uh, [nice...](https://bugs.python.org/issue14243#msg164504).